### PR TITLE
fix(curriculum): Add and modify tests to catch bad code.

### LINKED ISSUE #36414: Passed "React: Pass Props to a Stateless Functional Component" challenge with wrong solution
**Describe your problem and how to reproduce it:**
I passed the "React: Pass Props to a Stateless Functional Component" challenge with wrong solution as below, correct answer is commented beside:
```js 
const CurrentDate = (props) => {
  return (
    <div>
      { /* change code below this line */ }
      <p>The current date is: {Date()}</p>//<p>The current date is: {props.date}</p>
      { /* change code above this line */ }
    </div>
  );
};

class Calendar extends React.Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <div>
        <h3>What date is it?</h3>
        { /* change code below this line */ }
        <CurrentDate date="date"/>//<CurrentDate date={Date()} />
        { /* change code above this line */ }
      </div>
    );
  }
};
``` 

**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/front-end-libraries/react/pass-props-to-a-stateless-functional-component

**Tell us about your browser and operating system:**
* Browser Name:  Google Chrome
* Browser Version: Version 75.0.3770.100 (Official Build) (64-bit)
* Operating System: Windows 10

This is my first time opening an issue, I followed ["Creating a New Github Issue" guide on freeCodeCamp forum](https://www.freecodecamp.org/forum/t/creating-a-new-github-issue/18392). Please let me know if I am doing it wrong or there is a better approach, thank you.

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

As discussed in #36414 the challenge Pass Props to a Stateless Functional Component can be passed using code that does not follow the instructions.  This updates that challenge by requiring the value of the prop to be generated by `Date()` and checking that the `date` prop is used by `CurrentDate`.

Closes #36414

### CODE REVIEW

**Review by scissorsneedfoodtoo — APPROVED** [review]
Tested locally and everything LGTM. I like the way you wrote the test on line 46 to strip out all the white space before the regex test. I'll have to remember that for the new project based curriculum.

Will wait for another review before merging.

**Review by RandellDawson — APPROVED** [review]
Tested locally and everything works as intended.

**Comment by RandellDawson** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 🎉

**Pull request merged**
